lib/instrumental/agent.rb

```diff
@@ -365,9 +367,15 @@ def test_connection
     end
 
     def start_connection_worker
-      if enabled?
+      # NOTE: We need a mutex around both `running?` and thread creation,
+      # otherwise we could create two threads.
+      # Return early and queue the message if another thread is
```

Comment:
Since we don't do any message handling in this message, should we mention it in comments?